spring-grpc-spring-boot-autoconfigure/src/main/java/org/springframework/grpc/autoconfigure/server/GrpcServerFactoryAutoConfiguration.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2024-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.grpc.autoconfigure.server;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.AutoConfigureOrder;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnNotWebApplication;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.web.servlet.ServletRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.Ordered;
+
+import io.grpc.BindableService;
+import io.grpc.servlet.jakarta.GrpcServlet;
+
+/**
+ * {@link EnableAutoConfiguration Auto-configuration} for gRPC server-side components.
+ * <p>
+ * gRPC must be on the classpath and at least one {@link BindableService} bean registered
+ * in the context in order for the auto-configuration to execute.
+ *
+ * @author David Syer
+ * @author Chris Bono
+ */
+@AutoConfiguration
+@AutoConfigureOrder(Ordered.HIGHEST_PRECEDENCE)
+@ConditionalOnClass(BindableService.class)
+public class GrpcServerFactoryAutoConfiguration {
+
+	@Configuration(proxyBeanMethods = false)
+	@ConditionalOnNotWebApplication
```

Comment:
<sub><sub>**Legend:** `Tomcat/Jetty/Undertow` --> `T/J/UT`</sub></sub>
--------------

Before this change, we could run Spring gRPC in a servlet web app (T/J/UT) or Reactive web app (Netty). <sub>If its a Reactive web app it has to be Netty of course.</sub>

With this change, we can not run Spring gRPC in any webapp.

Should we only enforce a limitation when the user desires the servlet to be available? And even in that case, if the `server.port` and `spring.grpc.server.port` are different we should be able to run Spring gRPC and GrpcServlet in a servlet web app (T/J/UT) or in a reactive web app (Netty).

If we want the ports to be the same then the Spring gRPC server must be disabled which goes against the goal of the motivating issue #18 , serving both MVC and gRPC from same port (e.g. `9090`).

Reply:
> With this change, we can not run Spring gRPC in any webapp.

I don't think that's a very common scenario. I kind of expect that we will eventually support it through an `*.enabled` flag. I didn't want to do that until someone asks for it because it will make the `@Conditionals` a lot more complicated.